pkg/synchromanager/clustersynchro_manager.go

```diff
@@ -315,22 +315,24 @@ func (manager *Manager) UpdateClusterStatus(ctx context.Context, name string, st
 			return err
 		}
 
-		if equality.Semantic.DeepEqual(&cluster.Status, status) {
-			return nil
-		}
-
-		cluster = cluster.DeepCopy()
-
+		expectStatus := cluster.Status.DeepCopy()
 		if status.Version != "" {
-			cluster.Status.Version = status.Version
+			expectStatus.Version = status.Version
 		}
 		if status.SyncResources != nil {
-			cluster.Status.SyncResources = status.SyncResources
+			expectStatus.SyncResources = status.SyncResources
 		}
 		for _, condition := range status.Conditions {
-			meta.SetStatusCondition(&cluster.Status.Conditions, condition)
+			meta.SetStatusCondition(&expectStatus.Conditions, condition)
+		}
+
+		cluster = cluster.DeepCopy()
```

Comment:
It looks like it only needs to be deepcopied once.
eg.
```go

                lastStatus := cluster.Status
                cluster = cluster.DeepCopy()

                if status.Version != "" {
                        cluster.Status.Version = status.Version
                }
                if status.SyncResources != nil {
                        cluster.Status.SyncResources = status.SyncResources
                }
                for _, condition := range status.Conditions {
                        meta.SetStatusCondition(&cluster.Status.Conditions, condition)
                }

                if equality.Semantic.DeepEqual(cluster.Status, lastStatus) {
                        return nil
                }
```
What do you think?

Reply:
Exactly